### PR TITLE
SCR release v3.0, including components

### PR DESCRIPTION
**Pull request by gonsie**
Includes:
- SCR 3.0.0
- AXL 0.6.0
- ER 0.2.0
- KVTree 1.3.0
- Rankstr 0.2.0
- Redset 0.2.0
- Shuffile v0.2.0
- Spath v0.2.0

### CODE REVIEW

File: var/spack/repos/builtin/packages/axl/package.py [hunk 1]
```diff
@@ -36,6 +37,9 @@ class Axl(CMakePackage):
     depends_on('kvtree')
     depends_on('zlib', type='link')
 
+    depends_on('kvtree@main', when='@main')
+    depends_on('kvtree@1.3.0', when='@0.6.0')
+
```
Comment (CamStan) [inline var/spack/repos/builtin/packages/axl/package.py:43]: Will Spack users need access to the new `ENABLE_PTHREADS` cmake option, or is it not necessary here?

File: var/spack/repos/builtin/packages/er/package.py [hunk 2]
```diff
@@ -28,8 +29,13 @@ class Er(CMakePackage):
     depends_on('shuffile')
     depends_on('zlib', type='link')
 
-    variant('shared', default=True, description='Build with shared libraries')
+    depends_on('kvtree@1.3')
```
Comment (CamStan) [inline var/spack/repos/builtin/packages/er/package.py]: Does this need a `when='@0.2.0'`?

File: var/spack/repos/builtin/packages/scr/package.py [hunk 3]
```diff
@@ -32,6 +32,7 @@ class Scr(CMakePackage):
     version('develop', branch='develop')
     version('legacy',  branch='legacy')
 
+    version('3.0.0',  sha256='e204d3e99a49efac50b4bedc7ac05f55a05f1a65429500d919900c82490532cc', preferred=True)
     version('3.0rc2', sha256='4b2a718af56b3683e428d25a2269c038e9452db734221d370e3023a491477fad')
     version('3.0rc1', sha256='bd31548a986f050024429d8ee3644eb135f047f98a3d503a40c5bd4a85291308')
```
Comment (CamStan) [inline var/spack/repos/builtin/packages/scr/package.py]: Should we deprecate at least the oldest release candidate (for now) so users get a message if they try to use it? Also so we can eventually remove them and clean up our Spack package.

**Review by CamStan — COMMENTED** [review]
This is great @gonsie! Thank you! Just have a couple questions on some minor things.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
There have been a few fetching errors with packages in this PR. Rerunning to check if it was a temporary issue or not.

File: var/spack/repos/builtin/packages/axl/package.py (hunk 1 above)
Comment (gonsie) [reply var/spack/repos/builtin/packages/axl/package.py:43]: thanks @CamStan yes we need this here.

File: var/spack/repos/builtin/packages/er/package.py (hunk 2 above)
Comment (gonsie) [reply var/spack/repos/builtin/packages/er/package.py]: thank you!

File: var/spack/repos/builtin/packages/scr/package.py (hunk 3 above)
Comment (gonsie) [reply var/spack/repos/builtin/packages/scr/package.py]: good idea. thank you

**Review by CamStan — APPROVED** [review]

**Comment by gonsie** [comment]
@alalazo We've finally passed all the checks, I think this is ready for merging!

**Comment by gonsie** [comment]
ping @becker33 maybe you could help us out?

**Comment by gonsie** [comment]
pulling out the big guns. @tgamblin can you help merge this?

**Review by alalazo — APPROVED** [review]

**Pull request merged**